### PR TITLE
Feature/more auditing

### PR DESCRIPTION
**Pull request by MichaelRobinson-BJSS**
Added IAudited to outstanding relevant tables
Removed need to save changes as a specific user when seeding end to end seed data
Fixed migration package

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkEntityTypeConfiguration.cs [hunk 1]
```diff
@@ -19,6 +20,13 @@ public void Configure(EntityTypeBuilder<Framework> builder)
 
             builder.Property(f => f.Owner).HasMaxLength(100);
             builder.Property(f => f.ShortName).HasMaxLength(25);
+            builder.Property(f => f.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(f => f.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkEntityTypeConfiguration.cs:24]: don't think we need this property declaration if we're not configuring it.

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkSolutionEntityTypeConfiguration.cs [hunk 2]
```diff
@@ -18,6 +19,9 @@ public void Configure(EntityTypeBuilder<FrameworkSolution> builder)
                 .HasMaxLength(14)
                 .HasConversion(id => id.ToString(), id => CatalogueItemId.ParseExact(id));
 
+            builder.Property(f => f.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(f => f.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkSolutionEntityTypeConfiguration.cs:23]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/MarketingContactEntityTypeConfiguration.cs [hunk 3]
```diff
@@ -21,6 +22,8 @@ public void Configure(EntityTypeBuilder<MarketingContact> builder)
             builder.Property(m => m.FirstName).HasMaxLength(35);
             builder.Property(m => m.LastName).HasMaxLength(35);
             builder.Property(m => m.PhoneNumber).HasMaxLength(35);
+            builder.Property(m => m.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(m => m.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/MarketingContactEntityTypeConfiguration.cs:26]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/PricingUnitEntityTypeConfiguration.cs [hunk 4]
```diff
@@ -24,6 +25,14 @@ public void Configure(EntityTypeBuilder<PricingUnit> builder)
 
             builder.Property(u => u.TierName)
                 .HasMaxLength(30);
+
+            builder.Property(u => u.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(u => u.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/PricingUnitEntityTypeConfiguration.cs:30]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceAvailabilityTimesEntityTypeConfiguration.cs [hunk 5]
```diff
@@ -17,6 +18,9 @@ public void Configure(EntityTypeBuilder<ServiceAvailabilityTimes> builder)
                 .HasMaxLength(14)
                 .HasConversion(id => id.ToString(), id => CatalogueItemId.ParseExact(id));
 
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceAvailabilityTimesEntityTypeConfiguration.cs:22]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceLevelAgreementEntityTypeConfiguration.cs [hunk 6]
```diff
@@ -17,10 +18,18 @@ public void Configure(EntityTypeBuilder<ServiceLevelAgreements> builder)
                 .HasMaxLength(14)
                 .HasConversion(id => id.ToString(), id => CatalogueItemId.ParseExact(id));
 
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceLevelAgreementEntityTypeConfiguration.cs:22]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaContactEntityTypeConfiguration.cs [hunk 7]
```diff
@@ -13,6 +14,9 @@ public void Configure(EntityTypeBuilder<SlaContact> builder)
 
             builder.HasKey(s => s.Id);
 
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaContactEntityTypeConfiguration.cs:18]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaServiceLevelsEntityTypeConfiguration.cs [hunk 8]
```diff
@@ -17,10 +18,18 @@ public void Configure(EntityTypeBuilder<SlaServiceLevel> builder)
                 .HasMaxLength(14)
                 .HasConversion(id => id.ToString(), id => CatalogueItemId.ParseExact(id));
 
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaServiceLevelsEntityTypeConfiguration.cs:22]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SolutionEntityTypeConfiguration.cs [hunk 9]
```diff
@@ -22,6 +23,8 @@ public void Configure(EntityTypeBuilder<Solution> builder)
             builder.Property(s => s.FullDescription).HasMaxLength(3000);
             builder.Property(s => s.ImplementationDetail).HasMaxLength(1100);
             builder.Property(s => s.IntegrationsUrl).HasMaxLength(1000);
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SolutionEntityTypeConfiguration.cs:27]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardCapabilityEntityTypeConfiguration.cs [hunk 10]
```diff
@@ -10,6 +11,9 @@ public void Configure(EntityTypeBuilder<StandardCapability> builder)
         {
             builder.ToTable("StandardsCapabilities", Schemas.Catalogue);
 
+            builder.Property(sc => sc.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(sc => sc.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardCapabilityEntityTypeConfiguration.cs:15]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardEntityTypeConfiguration.cs [hunk 11]
```diff
@@ -32,9 +33,17 @@ public void Configure(EntityTypeBuilder<Standard> builder)
                 .IsRequired()
                 .HasColumnName("StandardTypeId");
 
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardEntityTypeConfiguration.cs:37]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierContactEntityTypeConfiguration.cs [hunk 12]
```diff
@@ -31,14 +32,17 @@ public void Configure(EntityTypeBuilder<SupplierContact> builder)
                 .IsRequired()
                 .HasMaxLength(6);
 
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierContactEntityTypeConfiguration.cs:36]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierEntityTypeConfiguration.cs [hunk 13]
```diff
@@ -29,8 +30,9 @@ public void Configure(EntityTypeBuilder<Supplier> builder)
 
             builder.Property(s => s.Summary).HasMaxLength(1100);
             builder.Property(s => s.SupplierUrl).HasMaxLength(1000);
-
             builder.Property(s => s.IsActive).HasDefaultValue(0);
+            builder.Property(s => s.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(s => s.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierEntityTypeConfiguration.cs:35]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierServiceAssociationEntityTypeConfiguration.cs [hunk 14]
```diff
@@ -23,11 +24,19 @@ public void Configure(EntityTypeBuilder<SupplierServiceAssociation> builder)
                 .HasMaxLength(14)
                 .HasConversion(id => id.ToString(), id => CatalogueItemId.ParseExact(id));
 
+            builder.Property(ssa => ssa.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(ssa => ssa.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierServiceAssociationEntityTypeConfiguration.cs:28]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ContactEntityTypeConfiguration.cs [hunk 15]
```diff
@@ -17,6 +18,13 @@ public void Configure(EntityTypeBuilder<Contact> builder)
             builder.Property(c => c.FirstName).HasMaxLength(100);
             builder.Property(c => c.LastName).HasMaxLength(100);
             builder.Property(c => c.Phone).HasMaxLength(35);
+            builder.Property(c => c.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(c => c.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ContactEntityTypeConfiguration.cs:22]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/DefaultDeliveryDateEntityTypeConfiguration.cs [hunk 16]
```diff
@@ -18,11 +19,18 @@ public void Configure(EntityTypeBuilder<DefaultDeliveryDate> builder)
                 .HasConversion(id => id.ToString(), id => CatalogueItemId.ParseExact(id));
 
             builder.Property(d => d.DeliveryDate).HasColumnType("date");
+            builder.Property(d => d.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(d => d.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/DefaultDeliveryDateEntityTypeConfiguration.cs:23]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/OrderItemRecipientEntityTypeConfiguration.cs [hunk 17]
```diff
@@ -18,6 +19,8 @@ public void Configure(EntityTypeBuilder<OrderItemRecipient> builder)
 
             builder.Property(r => r.OdsCode).HasMaxLength(8);
             builder.Property(r => r.DeliveryDate).HasColumnType("date");
+            builder.Property(r => r.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(r => r.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/OrderItemRecipientEntityTypeConfiguration.cs:23]: same as above

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ServiceRecipientEntityTypeConfiguration.cs [hunk 18]
```diff
@@ -14,6 +15,13 @@ public void Configure(EntityTypeBuilder<ServiceRecipient> builder)
 
             builder.Property(r => r.Name).HasMaxLength(256);
             builder.Property(r => r.OdsCode).HasMaxLength(8);
+            builder.Property(r => r.LastUpdated).HasDefaultValue(DateTime.UtcNow);
+            builder.Property(r => r.LastUpdatedBy);
```
Comment (leerogers-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ServiceRecipientEntityTypeConfiguration.cs:19]: same as above

**Review by leerogers-bjss — CHANGES_REQUESTED** [review]
looks good, just don't need the extra empty property definitions on the customization

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkEntityTypeConfiguration.cs (hunk 1 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkSolutionEntityTypeConfiguration.cs (hunk 2 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/FrameworkSolutionEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/MarketingContactEntityTypeConfiguration.cs (hunk 3 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/MarketingContactEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/PricingUnitEntityTypeConfiguration.cs (hunk 4 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/PricingUnitEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceAvailabilityTimesEntityTypeConfiguration.cs (hunk 5 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceAvailabilityTimesEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceLevelAgreementEntityTypeConfiguration.cs (hunk 6 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/ServiceLevelAgreementEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaContactEntityTypeConfiguration.cs (hunk 7 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaContactEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaServiceLevelsEntityTypeConfiguration.cs (hunk 8 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SlaServiceLevelsEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SolutionEntityTypeConfiguration.cs (hunk 9 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SolutionEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardCapabilityEntityTypeConfiguration.cs (hunk 10 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardCapabilityEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardEntityTypeConfiguration.cs (hunk 11 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/StandardEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierContactEntityTypeConfiguration.cs (hunk 12 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierContactEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierServiceAssociationEntityTypeConfiguration.cs (hunk 14 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierServiceAssociationEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ContactEntityTypeConfiguration.cs (hunk 15 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ContactEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierEntityTypeConfiguration.cs (hunk 13 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Catalogue/Configuration/SupplierEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/DefaultDeliveryDateEntityTypeConfiguration.cs (hunk 16 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/DefaultDeliveryDateEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/OrderItemRecipientEntityTypeConfiguration.cs (hunk 17 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/OrderItemRecipientEntityTypeConfiguration.cs]: Done

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ServiceRecipientEntityTypeConfiguration.cs (hunk 18 above)
Comment (MichaelRobinson-BJSS) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/ServiceRecipientEntityTypeConfiguration.cs]: Done

**Review by leerogers-bjss — APPROVED** [review]

**Pull request merged**